Make methods of internal file_obj visible from StreamWrapper

ejguan opened the pull request:
Fixes #71221

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#71653 Make methods of internal file_obj visible from StreamWrapper**

Differential Revision: [D33718749](https://our.internmc.facebook.com/intern/diff/D33718749)

ejguan committed:
Make methods of internal file_obj visible from StreamWrapper

[ghstack-poisoned]

ejguan committed:
Update on "Make methods of internal file_obj visible from StreamWrapper"




[ghstack-poisoned]

ejguan commented:
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33718749).

ejguan committed:
Update on "Make methods of internal file_obj visible from StreamWrapper"

Fixes #71221


Differential Revision: [D33718749](https://our.internmc.facebook.com/intern/diff/D33718749)

[ghstack-poisoned]

ejguan commented:
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33718749).

ejguan commented on torch/utils/data/datapipes/utils/common.py:
```diff
@@ -99,29 +99,41 @@ def deprecation_warning(name, new_name: str = ""):
 
 
 class StreamWrapper:
```
I was planning to make `isinstance` check also working. But, after a while of research, this may not be possible because the `file_obj` can be arbitrary python file-like object. It's almost impossible for me to add a metaclass to handle all cases.

pmeier commented on test/test_datapipe.py:
```diff
@@ -170,6 +171,62 @@ def test_random_shuffle(self):
 
 
+class TestStreamWrapper(TestCase):
+    class _FakeFD:
```
Could you also open a test for `open`?

NivekT reviewed (approved):
LGTM. The CI failures seem unrelated but we probably need to rebase before landing.

ejguan committed:
Update on "Make methods of internal file_obj visible from StreamWrapper"

Fixes #71221


Differential Revision: [D33718749](https://our.internmc.facebook.com/intern/diff/D33718749)

[ghstack-poisoned]

ejguan commented:
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33718749).

pmeier commented on torch/utils/data/datapipes/utils/common.py:
```diff
@@ -99,29 +99,41 @@ def deprecation_warning(name, new_name: str = ""):
-    def __setstate__(self, state):
-        self.file_obj.__setstate__(state)
+    def __repr__(self):
+        return "StreamWrapper<" + repr(self.file_obj) + ">"
```
Nit

```suggestion
        return f"StreamWrapper<{self.file_obj!r}>"
```

ejguan committed:
Update on "Make methods of internal file_obj visible from StreamWrapper"

Fixes #71221


Differential Revision: [D33718749](https://our.internmc.facebook.com/intern/diff/D33718749)

[ghstack-poisoned]

ejguan commented:
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33718749).
